radix-engine/src/transaction/state_update_summary.rs

```diff
@@ -76,11 +79,16 @@ impl Add for BalanceChange {
 
     fn add(mut self, rhs: Self) -> Self::Output {
         match &mut self {
-            BalanceChange::Fungible(self_value) => {
-                let BalanceChange::Fungible(value) = rhs else {
+            BalanceChange::Fungible {
+                change: self_change,
+                resultant_balance: self_resultant_balance,
+            } => {
+                let BalanceChange::Fungible{ change, resultant_balance } = rhs else {
                     panic!("cannot {:?} + {:?}", self, rhs);
                 };
-                *self_value = self_value.safe_add(value).unwrap();
+                *self_change = self_change.safe_add(change).unwrap();
+                *self_resultant_balance =
```

Comment:
This isn't correct.

The Resultant Balance is only valid if we know statically there's only a single vault of the given type under the given component.

So EG it should only be added for Accounts, for vaults in the account vault partition.

Reply:
I think there is a confusion between "vault balance" vs "account balance".

Engine only collects and returns "vault balance changes".
Node aggregates them into "account balance changes".

The piece of code commented here is just a `impl Add for BalanceChange`, which is actually not used in the Engine (at least not for aggregating "account balance changes"), but is used in the Node.